packages/contracts-bedrock/scripts/upgrades/SafeBuilder.sol

```diff
@@ -57,6 +57,37 @@ abstract contract SafeBuilder is Script {
         return success;
     }
 
+    /**
+     * @notice Computes the safe transaction hash.
+     */
+    function _computeSafeTransactionHash(address _safe, address _proxyAdmin) public returns (bytes32) {
+        // Ensure that the required contracts exist
+        require(address(multicall).code.length > 0, "multicall3 not deployed");
+        require(_safe.code.length > 0, "no code at safe address");
+        require(_proxyAdmin.code.length > 0, "no code at proxy admin address");
+
+        IGnosisSafe safe = IGnosisSafe(payable(_safe));
+        uint256 nonce = safe.nonce();
```

Comment:
This is good but note that there is a race condition here, ie the wrong nonce will be grabbed if another tx frontruns this one. Perhaps a way around that would be to use ffi to `eth_getLogs` and figure out what nonce to use, altho might be annoying

Reply:
Might require brute forcing and checking to see with `eth_call` to see which hash is in the mapping then use that nonce

Reply:
Discussion was had offline; closing since we settled on not over-engineering against nonce race conditions for now.